pkg/storage/utils/decomposedfs/decomposedfs.go

```diff
@@ -228,23 +228,32 @@ func (fs *Decomposedfs) Postprocessing(ch <-chan interface{}) {
 			); err != nil {
 				log.Error().Err(err).Str("uploadID", ev.UploadID).Msg("Failed to publish UploadReady event")
 			}
-		case events.VirusscanFinished:
-			if ev.Error != nil {
-				// scan failed somehow
-				continue
-			}
-
-			up, err := upload.Get(ctx, ev.UploadID, fs.lu, fs.tp, fs.o.Root, fs.stream, fs.o.AsyncFileUploads, fs.o.Tokens)
-			if err != nil {
-				log.Error().Err(err).Str("uploadID", ev.UploadID).Msg("Failed to get upload")
-				continue
-			}
-
-			if err := up.Node.SetScanData(ev.Description, ev.Scandate); err != nil {
-				log.Error().Err(err).Str("uploadID", ev.UploadID).Msg("Failed to set scan results")
-				continue
-			}
-
+		default:
+			log.Error().Interface("event", ev).Msg("Unknown event")
+
+			// NOTE: Listening to the VirusscanFinished event enables the server to store the "lastScanned" date
+			// For a nice UX this should be displayed by the user.
+			// However enabling it requires the decision if the etag should change after a virusscan (and proper testing to verify this)
```

Comment:
Currently an ETag change would cause a re-download of the file for most clients. We need a solution for metadata-only changes, but currently we are not there. Compare this fairly old issue: https://github.com/owncloud/client/issues/4755

I'd say we stay with the current approach to NOT change the ETag after this kind of metadata has changed.

Reply:
I'm afraid that is NOT the current approach. The etag is calculated from the mtime of the file. If any extended attribute is added the mtime is changed and therefore the etag changes.

Reply:
Not true. When *only* an xattr is added or changed, the mtime is NOT changed, but only the ctime! As long as we only look on the mtime, there would not be a change.

Reply:
Reenabled functionality after yesterdays discovery.